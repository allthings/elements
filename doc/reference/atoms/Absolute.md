# Absolute

## Usage
| Name        | Type           | Description  |
| ----------- |:--------------:| ------------:|
|children|node|Absolute children
|top|union|Top offset
|bottom|union|Bottom offset
|left|union|Left offset
|right|union|Right offset
