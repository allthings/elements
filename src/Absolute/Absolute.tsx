import React, { FunctionComponent } from 'react'
import View, { IViewProps } from '../View'
import { css } from 'glamor'

const absolute = ({ left, right, top, bottom }: IAbsoluteProps) =>
  css({
    position: 'absolute',
    top,
    right,
    left,
    bottom,
  })

export interface IAbsoluteProps {
  /** Top offset */
  top?: string | number
  /** Bottom offset */
  bottom?: string | number
  /** Left offset */
  left?: string | number
  /** Right offset */
  right?: string | number
}

const Absolute: FunctionComponent<IAbsoluteProps & IViewProps> = ({
  top,
  left,
  right,
  bottom,
  children,
  ...props
}) => (
  <View {...absolute({ top, left, right, bottom })} {...props}>
    {children}
  </View>
)

export default Absolute
