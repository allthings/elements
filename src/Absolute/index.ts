export { default } from './Absolute'
export * from './Absolute'
