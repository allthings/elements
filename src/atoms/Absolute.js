import React from 'react'
import PropTypes from 'prop-types'
import View from '../atoms/View'
import { css } from 'glamor'

const absolute = ({ left, right, top, bottom }) =>
  css({
    position: 'absolute',
    top,
    right,
    left,
    bottom,
  })

class Absolute extends React.Component {
  static propTypes = {
    /** Absolute children */
    children: PropTypes.node,
    /** Top offset */
    top: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    /** Bottom offset */
    bottom: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    /** Left offset */
    left: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    /** Right offset */
    right: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  }

  render() {
    const { top, left, right, bottom, children, ...props } = this.props
    return (
      <View {...absolute({ top, left, right, bottom })} {...props}>
        {children}
      </View>
    )
  }
}

export default Absolute
