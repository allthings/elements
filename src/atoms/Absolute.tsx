import React from 'react'
import View, { IViewProps } from '../atoms/View'
import { css } from 'glamor'

const absolute = ({ left, right, top, bottom }: IProps) =>
  css({
    position: 'absolute',
    top,
    right,
    left,
    bottom,
  })

interface IProps {
  /** Top offset */
  top?: string | number
  /** Bottom offset */
  bottom?: string | number
  /** Left offset */
  left?: string | number
  /** Right offset */
  right?: string | number
}

const Absolute = ({
  top,
  left,
  right,
  bottom,
  children,
  ...props
}: IProps & IViewProps) => (
  <View {...absolute({ top, left, right, bottom })} {...props}>
    {children}
  </View>
)

export default Absolute
