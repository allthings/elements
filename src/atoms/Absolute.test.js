import React from 'react'
import renderer from 'react-test-renderer'
import Absolute from './Absolute'

test('Absolute renders without error', () => {
  const tree = renderer.create(
    <Absolute top={3} left={2} right={1} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
