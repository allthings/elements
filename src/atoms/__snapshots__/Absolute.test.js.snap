// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Absolute renders without error 1`] = `
.css-17z7rhr,
[data-css-17z7rhr] {
  position: absolute;
  top: 3px;
  right: 1px;
  left: 2px;
}

.css-4p3ux5,
[data-css-4p3ux5] {
  box-sizing: border-box;
  flex: 0 0 auto;
  -webkit-flex: 0 0 auto;
}

<div
  data-css-17z7rhr=""
  data-css-4p3ux5=""
/>
`;
